### PR TITLE
New: string and json type input validation checks are added

### PR DESCRIPTION
**Pull request by johnymathew**
The string validator `@sv(min: Int, max: Int)` on any input field will check for min and max length and illegal characters.
if you just provide `@sv` it will just check for allowed characters and assume the default min and max lengths are 1 and 256
The legal characters for string are a-z, A-Z, 0-9, -, _, / and few others .  

`@jv` is the json validator on total elements and key/value length if they are string type.

### CODE REVIEW

File: app/apollo/schema/channel.js [hunk 1]
```diff
@@ -69,60 +69,60 @@ const channelSchema = gql`
...
      Adds a channel
      """
-     addChannel(orgId: String!, name: String!): AddChannelReply!
+     addChannel(orgId: String! @sv, name: String! @sv): AddChannelReply!
      
      """
      Edits a channel
      """
-     editChannel(orgId: String!, uuid: String!, name: String!): EditChannelReply!
+     editChannel(orgId: String! @sv, uuid: String!, name: String! @sv): EditChannelReply!
```
[the remark this answers is not in the document]
Comment (johnymathew) [reply app/apollo/schema/channel.js:109]: fixed

File: app/apollo/schema/group.js [hunk 2]
```diff
@@ -54,59 +54,59 @@ const groupSchema = gql`
...
     unGroup a list of clusters
     """
-    unGroupClusters(orgId: String! uuid: String! clusters: [String]!): UnGroupClustersReply!
+    unGroupClusters(orgId: String! @sv uuid: String! @sv clusters: [String]! @sv): UnGroupClustersReply!
 
     """
     Adds a list of groups to a list of clusterIds
     """
-    assignClusterGroups(orgId: String!, groupUuids: [String!]!, clusterIds: [String!]!): AssignClusterGroupsReply!
+    assignClusterGroups(orgId: String! @sv, groupUuids: [String!]! @sv, clusterIds: [String!]! @sv): AssignClusterGroupsReply!
```
[the remark this answers is not in the document]
Comment (johnymathew) [reply app/apollo/schema/group.js:99]: yes

File: app/apollo/utils/directives.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    }
+    field.sanitizers.push(sanitizer);
+  }
+}
+
+
+class JsonSanitizer extends Sanitizer {
+
+  constructor(arg) {
+    super('Identifer_', arg);
```
[the remark this answers is not in the document]
Comment (johnymathew) [reply app/apollo/utils/directives.js:104]: changed

File: app/apollo/utils/directives.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  }
+
+  parseTree(parent) {
+    var hasNonLeafNodes = false;
+    var childCount = 0;  
+    for (var child in parent) {
+      if (typeof parent[child] === 'object') {
```
[the remark this answers is not in the document]
Comment (johnymathew) [reply app/apollo/utils/directives.js:111]: handled as part of string array

File: app/apollo/utils/directives.js [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      }
+    }
+
+  }
+
+  validateSting(value) {
+    var MAXLEN = 256;
+    var MINLEN = 1;
+    const pattern = /[<>$%&!#]{1,}/;
+    if (this.arg === 'content')  MAXLEN = 10000;
```
[the remark this answers is not in the document]
Comment (johnymathew) [reply app/apollo/utils/directives.js:60]: sonw

**Comment by yingwang-us** [comment]
@mckaymic @dalehille please also help review. thanks.

**Pull request merged**